admiral/pkg/controller/admiral/routingpolicy_test.go

```diff
@@ -0,0 +1,97 @@
+package admiral
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/istio-ecosystem/admiral/admiral/pkg/apis/admiral/model"
+	v1 "github.com/istio-ecosystem/admiral/admiral/pkg/apis/admiral/v1"
+	"github.com/istio-ecosystem/admiral/admiral/pkg/test"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"testing"
+	"time"
+)
+
+
+func TestNewroutingPolicyController(t *testing.T) {
+	config, err := clientcmd.BuildConfigFromFlags("", "../../test/resources/admins@fake-cluster.k8s.local")
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+	stop := make(chan struct{})
+	handler := test.MockRoutingPolicyHandler{}
+
+	routingPolicyController, err := NewRoutingPoliciesController(stop, &handler, config, time.Duration(1000))
+
+	if err != nil {
+		t.Errorf("Unexpected err %v", err)
+	}
+
+	if routingPolicyController == nil {
+		t.Errorf("RoutingPolicy controller should never be nil without an error thrown")
+	}
+}
+
+
+func TestRoutingPolicyAddUpdateDelete(t *testing.T) {
```

Comment:
Will be good to use table tests as we are testing 3 functionalities.

Reply:
will update

Reply:
single test has add/update and delete they are dependent